ui/src/components/auth/ResetPasswordContainer.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useState } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import axios from 'axios';
+import {
+  Box,
+  Button,
+  Stack,
+  TextField,
+  Typography,
+  Snackbar,
+  Alert,
+  AlertColor,
+} from '@mui/material';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { ResetPasswordFields } from './helpers/form.types';
+import { formSpacing, formWidth } from './helpers/constants';
+import { useLocation } from 'react-router-dom';
+import { stringValidator, passwordValidator } from './helpers/zod.validators';
+
+// zod validation
+const resetPwdValidationSchema = z
+  .object({
+    password: passwordValidator,
+    confirmPassword: stringValidator,
+  })
+  .refine(
+    (values) => {
+      return values.password === values.confirmPassword;
+    },
+    {
+      message: 'The passwords do not match',
+      path: ['confirmPassword'],
+    }
+  );
+
+const useQuery = () => {
+  return new URLSearchParams(useLocation().search);
+};
+
+const ResetPasswordContainer: React.FC = (props: any) => {
+  const { history } = props;
```

Comment:
I dont think having a parameter in a container is a good idea. Esp since u cant pass any params to it cuz its directly linked to routes.

I saw ur using history to push the link, maybe u can use smth like this from the main web?
![Screenshot_20231129_193034_GitHub.jpg](https://github.com/hmcc-global/psalted-2/assets/56828805/6bc5cfc6-bb2c-447a-b411-0e46bfc384d9)